server/src/main/java/io/druid/server/http/NamespacesResource.java

```diff
@@ -0,0 +1,242 @@
+    final ObjectMapper mapper = isSmile ? smileMapper : jsonMapper;
+
+    Response.ResponseBuilder builder = Response.status(Response.Status.ACCEPTED);
+
+
+    final Map<String, Object> map;
+    try {
+      map = mapper.readValue(
+          inputStream, new TypeReference<Map<String, Object>>()
+          {
+          }
+      );
+    }
+    catch (IOException e) {
+      log.error(e, "Error parsing new namespace");
+      return Response.status(Response.Status.BAD_REQUEST).entity(cleanError(e)).build();
+    }
+
+    try {
+      if (!map.containsKey("namespace")) {
+        return Response
+            .status(Response.Status.BAD_REQUEST)
+            .entity(cleanError(new IAE("`namespace` field not found")))
+            .build();
+      }
+      final Object nsObj = map.get("namespace");
+      if (!(nsObj instanceof String)) {
+        return Response.status(Response.Status.BAD_REQUEST).entity(
+            cleanError(
+                new IAE(
+                    String.format(
+                        "`namespace` field is not String, is [%s]", nsObj.getClass().getCanonicalName()
+                    )
+                )
+            )
+        ).build();
+      }
+      final String namespace = (String) nsObj;
+      final String path = ZKPaths.makePath(zkPathsConfig.getNamespacePath(), namespace);
+      if(curator.checkExists().forPath(path) == null) {
+        curator
+            .inTransaction()
+            .create()
+            .withMode(CreateMode.PERSISTENT)
+            .forPath(path, mapper.writeValueAsBytes(map))
+            .and()
+            .commit();
+      } else {
+        curator
+            .inTransaction()
+            .setData()
+            .forPath(path, mapper.writeValueAsBytes(map))
+            .and()
+            .commit();
+      }
+    }
+    catch (Exception e) {
+      log.error(e, "Error creating new namespace");
+      return Response.serverError().entity(cleanError(e)).build();
+    }
+    builder.entity(map);
+    return builder.build();
+  }
+
+  @DELETE
+  @Produces({MediaType.APPLICATION_JSON, SmileMediaTypes.APPLICATION_JACKSON_SMILE})
+  @Path("/{namespace}")
+  public Response deleteNamespace(@PathParam("namespace") String namespace)
+  {
+    if (Strings.isNullOrEmpty(namespace)) {
+      return Response.status(Response.Status.BAD_REQUEST)
+                     .entity(cleanError(new IAE("Must specify `namespace`")))
+                     .build();
+    }
+    final String path = ZKPaths.makePath(zkPathsConfig.getNamespacePath(), namespace);
+    try {
+      if (curator.checkExists().forPath(path) == null) {
+        return Response.status(Response.Status.NOT_FOUND)
+                       .entity(String.format("Namespace [%s] not found", namespace))
+                       .build();
+      }
+      curator.inTransaction().delete().forPath(path).and().commit();
+    }
+    catch (Exception e) {
+      if (e instanceof IAE) {
```

Comment:
this catch seems to only surround curator calls, I don't believe curator throws IAE instances...